x-pack/filebeat/input/streaming/websocket.go

```diff
@@ -82,48 +89,151 @@ func (s *websocketStream) FollowStream(ctx context.Context) error {
 	}
 
 	// websocket client
-	headers := formHeader(s.cfg)
-	c, resp, err := websocket.DefaultDialer.DialContext(ctx, url, headers)
+	c, resp, err := connectWebSocket(ctx, s.cfg, url, s.log)
+	handleConnectionResponse(resp, s.log)
+	if err != nil {
+		s.metrics.errorsTotal.Inc()
+		s.log.Errorw("failed to establish websocket connection", "error", err)
+		return err
+	}
+	// ensures this is the last connection closed when the function returns
+	defer c.Close()
+
+	for {
+		select {
+		case <-ctx.Done():
+			s.log.Debugw("context cancelled, closing websocket connection")
+			c.Close()
```

Comment:
`c.Close` is already in a `defer` statement, you can just return here. It will also avoid calling it twice.
```suggestion
```